Source/Core/DolphinQt/Config/Mapping/MappingWindow.cpp

```diff
@@ -348,7 +349,8 @@ void MappingWindow::SetMappingType(MappingWindow::Type type)
     widget = new WiimoteEmuGeneral(this, extension);
     setWindowTitle(tr("Wii Remote %1").arg(GetPort() + 1));
     AddWidget(tr("General and Options"), widget);
-    AddWidget(tr("Motion Controls"), new WiimoteEmuMotionControl(this));
+    AddWidget(tr("Motion Controls (Emulated)"), new WiimoteEmuMotionControl(this));
+    AddWidget(tr("Motion Controls (Real)"), new WiimoteEmuMotionControlIMU(this));
```

Comment:
Not sure how this shows up in the UI (haven't tested yet) but should it only be added if the controller actually supports that functionality?  That would split our interface up but I also feel like it will be confusing as is to many users who don't have those peripherals.  Maybe this could be moved out to a separate selection?  So we have a drop down that currently has:

* None
* Emulated Wiimote
* Real Wiimote

We could instead have:

* None
* Emulated Wiimote
* Emulated Wiimote from Motion Device

(names could be different)

That leads to my second comment...I can't think of a better name but "real" seems like it'd be confusing too.  This functionality is still emulated just so happens to be a closer emulation because the emulated device supports actual motion.

Reply:
I'm not sure I like the "Emulated" and "Real" either, but it wouldn't make sense to have the option in that drop down menu. And there is not really a reliable way to know if a controller provides raw accel/gyro data.

Reply:
Perhaps leave the "Motion Controls" tab named as-is and place the new "raw" mappings under an "Advanced" tab.

Reply:
> I'm not sure I like the "Emulated" and "Real" either, but it wouldn't make sense to have the option in that drop down menu

Yeah, the UI is already pretty substantial, I just worry about having another piece that would confuse users but I don't deal with them on a daily basis like @JMC47 or others do, so maybe they'd be ok with it.

Putting in the drop down just seemed to make the distinction a little clearer but it's not ideal either.

> And there is not really a reliable way to know if a controller provides raw accel/gyro data.

If they choose "Real" in this PR, how does Dolphin know it's receiving data?  Could we use similar data to determine detection?

Reply:
If we add a new item to the dropdown as you suggest, wouldn't that preclude the combination of "emulated" and "real" motion sources, as was just implemented?

The only other name I have been able to think of is Inertial Measurement Unit (IMU), which really is the correct name for this kind of motion source. But it seemed a bit heavy for the user interface, therefore "real". Some like to use "gyro", but that is really just one part of the IMU. But absolute correctness is perhaps not so important?

Reply:
> If we add a new item to the dropdown as you suggest, wouldn't that preclude the combination of "emulated" and "real" motion sources, as was just implemented?

Not necessarily.  Whatever distinction we used for the IMU based controller by definition means that it accepts either button presses or "real" motion.  At least that's how I viewed it.

> The only other name I have been able to think of is Inertial Measurement Unit (IMU), which really is the correct name for this kind of motion source. But it seemed a bit heavy for the user interface, therefore "real". Some like to use "gyro", but that is really just one part of the IMU. But absolute correctness is perhaps not so important?

Yeah unfortunately I don't have any better name at the moment.  If "Real" is all we can come up with, so be it, it just doesn't sit well.  Especially since we've been using "Real" to distinguish between a real wiimote vs emulated controller

Reply:
> If they choose "Real" in this PR, how does Dolphin know it's receiving data? Could we use similar data to determine detection?

I assume there is real data if it contains anything other than zero's. Perhaps hide the `Motion Controls (Real)` tab until data is detected?

Reply:
> Yeah unfortunately I don't have any better name at the moment. If "Real" is all we can come up with, so be it, it just doesn't sit well with me. Especially since we've been using "Real" to distinguish between a real wiimote vs an emulated one.

Then what about just "Gyro" (even though it's really just one part of the IMU)?

Reply:
> I assume there is real data if it contains anything other than zero's. Perhaps hide the Motion Controls (Real) tab until data is detected?

Possibly.

Although, I was just thinking about what a "guided" solution would look like.  You first click on "Motion Controls" and there's a question 'What type of motion input do you want to use?'  With two buttons:  'use only button presses', 'use button presses or motion from a gyro based device'.  If you click the first, you get a layout like is in master.  If you click the second, a prompt comes up asking you to perform a motion action to ensure you have a proper device connected.  If you don't, it would ask you to connect a device or switch to a button only mode.  That's more involved but would be nice..

The only thing with that is how to reset or change (if you had say a 360pad but then bought a DS4 and wanted to update the motion based controls..)

Reply:
> Although, I was just thinking about what a "guided" solution would look like....

That does seem a bit convoluted.

Currently, the only thing you need to do for "real" motion controls is select an `UDPServer` input device in the mapping window. If you select any other device you won't even know there is such functionality, except for the existence of the `Motion Controls (Real)` tab.

Reply:
This is how it currently looks. The contents of the other tabs are unchanged.

![2019-09-15 21_12_59-vs2017 (1) - Remote Viewer](https://user-images.githubusercontent.com/47765059/64926437-b41ce100-d7fd-11e9-8102-0302173c803c.png)

Reply:
IMO the current UI shown in this recent screenshot is not adequate to deter users from incorrectly trying to map traditional controller inputs to the "raw" mappings. Many users are "smart" enough to know what "Yaw Left" means but won't understand if they bind it to a button that they aren't gonna have a good time.

Reply:
> IMO the current UI shown in this recent screenshot is not adequate to deter users from incorrectly trying to map traditional controller inputs to the "raw" mappings.

I considered not actually showing these mappings, as they don't need modification with UDPServer input devices. I however left them in in the case there were other motion sources - and you showed that the case is so on Linux.

If hiding the mappings is not an option, perhaps a stern warning to the user that the boxes are not intended for regular button mapping is adequate?

Reply:
Basically the same screenshot as earlier, just changed the titles of the motion tabs. The `Motion Controls (Emulated)` tab is reverted to `Motion Controls`. The `Motion Controls (Real)` tab is renamed to `Motion Source`. Does this name work?

One could dynamically enable/disable (gray out) the `Motion Source` tab depending on if an input device is bound to the accelerometer and gyro mappings. This would work with all UDPServer input devices, since the accelerometer and gyro inputs always have the same names. However, this would preclude mapping of accelerometer and gyro inputs from non-UDPServer sources, like @jordan-woyak has done on Linux.

Another point in favor of not disabling the `Motion Source` tab is that however much the user screws up the settings there, it won't affect anything but "real" motion controls. Which won't matter if they don't have a "real" motion source.

![2019-09-16 02_17_53-vs2017 (1) - Remote Viewer](https://user-images.githubusercontent.com/47765059/64929630-bea0a000-d828-11e9-894d-a1363edc350f.png)

Reply:
Just a mockup with some warning text.

![2019-09-16 02_17_53-vs2017 (1) - Remote Viewer](https://user-images.githubusercontent.com/47765059/64930765-f7924200-d833-11e9-8c3d-a87847080b90.png)

Reply:
Here is my final suggestion.

![2019-09-16 20_01_36-vs2017 (1) - Remote Viewer](https://user-images.githubusercontent.com/47765059/64981845-34564b80-d8bd-11e9-8259-eebb5188ce84.png)

Reply:
And here's my final FINAL suggestion (sorry).

(Renamed `Motion Controls` tab to `Motion Simulation`.)

![2019-09-17 01_12_16-vs2017 (1) - Remote Viewer](https://user-images.githubusercontent.com/47765059/64999615-c889d800-d8e8-11e9-86a9-ff9ea6035d9e.png)

Reply:
I'm ok with this design, the warning isn't really in line with the rest of the UI but I think it is appropriate. I do think the Motion Simulation and Motion Input shouldn't be split by Extension as they have the same functionality.

Reply:
> I do think the Motion Simulation and Motion Input shouldn't be split by Extension as they have the same functionality.

I agree, fixed. Here's a screenshot:

![2019-09-19 15_04_21-vs2017 (1) - Remote Viewer](https://user-images.githubusercontent.com/47765059/65246630-428bae00-daef-11e9-934e-cdf96a498546.png)